src/notification/notification.vala

```diff
@@ -316,10 +319,14 @@ namespace SwayNotificatonCenter {
                     timeout = timeout_delay * 1000;
                     break;
                 case UrgencyLevels.CRITICAL:
-                    return;
+                    if (timeout_critical_delay == 0) {
+                        return;
+                    }
+                    timeout = timeout_critical_delay * 1000;
+                    break;
             }
             uint ms = param.expire_timeout > 0 ? param.expire_timeout : timeout;
-            if (param.expire_timeout != 0) {
+            if (ms != 0) {
```

Comment:
I'm not sure about this change. The previous check looked like a logic error to me.

Reply:
Woops. Yeah, it should be `if (ms != 0) {`